db/db_impl/db_impl.cc

```diff
@@ -1269,6 +1331,51 @@ Status DBImpl::ApplyReplicationLogRecord(ReplicationLogRecord record,
   return s;
 }
 
+Status DBImpl::GetReplicationRecordDebugString(
+    const ReplicationLogRecord& record, std::string* out) const {
+  std::ostringstream oss;
+
+  auto s = Status::OK();
+  switch (record.type) {
+    case ReplicationLogRecord::kMemtableWrite: {
+      if (record.contents.size() < 8) {
+        s = Status::Corruption("corrupted kMemtableWrite record");
+        break;
+      }
+      auto seq = DecodeFixed64(record.contents.data());
+      oss << "kMemtableWrite " << record.contents.size() << " bytes, sequence=" << seq;
+      break;
+    }
+    case ReplicationLogRecord::kMemtableSwitch: {
+      WriteContext write_context;
+      MemTableSwitchRecord mem_switch_record;
+      Slice contents_slice(record.contents);
+      s = DeserializeMemTableSwitchRecord(&contents_slice, &mem_switch_record);
+      if (!s.ok()) {
+        return s;
+      }
+      oss << "kMemtableSwitch next_log_num=" << mem_switch_record.next_log_num;
+      break;
+    }
+    case ReplicationLogRecord::kManifestWrite: {
+      Slice src = record.contents;
```

Comment:
Are we explicitly trying to use different `Slice` constructor between here and above with `Slice contents_slice(record.contents);`?